consoles/sshVirtsh.pm

```diff
@@ -325,111 +326,126 @@ sub add_interface {
     return;
 }
 
-sub add_disk {
-    my ($self, $args) = @_;
+sub to_byte {
+    my $sp              = shift;
+    my $size_multiplier = {
+        b => 1,
+        k => 1024, K => 1024,
+        M => 1024 * 1024,
+        G => 1024 * 1024 * 1024,
+        T => 1024 * 1024 * 1024 * 1024,
+    };
 
-    my $backingfile             = $args->{backingfile};
-    my $cdrom                   = $args->{cdrom};
-    my $name                    = $self->name;
-    my $file                    = $name . $args->{dev_id} . ($self->vmm_family eq 'vmware' ? '.vmdk' : '.img');
-    my $basedir                 = '/var/lib/libvirt/images/';
-    my $vmware_datastore        = get_var('VMWARE_DATASTORE', '');
-    my $vmware_openqa_datastore = "/vmfs/volumes/$vmware_datastore/openQA/";
-    if ($args->{create}) {
-        my $size = $args->{size} || '20G';
-        if ($self->vmm_family eq 'vmware') {
-            my $vmware_disk_path = $vmware_openqa_datastore . $file;
-            # Power VM off, delete it's disk image, and create it again.
-            # Than wait for some time for the VM to *really* turn off.
-            my $cmd =
-              "( set -x; vmid=\$(vim-cmd vmsvc/getallvms | awk \'/$name/ { print \$1 }\');" .
-              'if [ $vmid ]; then ' .
-              'vim-cmd vmsvc/power.off $vmid;' .
-              'vim-cmd vmsvc/destroy $vmid;' .
-              'fi;' .
-              "vmkfstools -v1 -U $vmware_disk_path;" .
-              "vmkfstools -v1 -c $size --diskformat thin $vmware_disk_path; sleep 10 ) 2>&1";
-            my $retval = $self->run_cmd($cmd, domain => 'sshVMwareServer');
-            die "Can't create VMware image $vmware_disk_path" if $retval;
+    #in case of invalid format return error
+    if ($sp !~ /^(?<num>\d+)(?<unit>[bkKMGT])?$/) {
+        return undef;
+    } else {
+        return $+{num} * $size_multiplier->{$+{unit} // 'b'};
+    }
+}
+
+sub gen_kvm_img {
+    my ($self, $args) = @_;
```

Comment:
In general I like such pattern more
```suggestion
    my ($self, %args) = @_;
```

With this you would call the method like:
```perl
$self->gen_kvm_img(new_image_file => 'foo.bla', size => '10G'); 
```
instead of 
$self->gen_kvm_img({new_image_file => 'foo.bla', size => '10G'});

Reply:
oh, my fault. I am on purpose buidling the hash, thus there is only one call of `gen_kvm_img`